src/Components/App/App.js

```diff
@@ -40,31 +43,45 @@ class App extends Component {
           }
           return product;
         });
+        const updatedFavorites = updatedProducts.filter(product => product.favorite);
 
-        this.setState({ products: updatedProducts});
+        this.setState({ products: updatedProducts, favorites: updatedFavorites});
       })
       .catch(error => {
         this.setState({ error: error.message });
       });
   }
 
-
   render = () => {
     return (
       this.state.error !== null ?
         <Error error={this.state.error} />
       :
-      <Router>
-        <main>
-          <Aside />
-          <Header />
-          <ProductsContainer 
-            products={this.state.products} 
-            addFavorite={this.addFavorite} 
+    <>
+      <Aside />
+      <Header />  
+      <Routes>
```

Comment:
I really like that we set these components so that they're always rendered regardless of the route.